src/battle_script_commands.c

```diff
@@ -7118,6 +7133,104 @@ static void Cmd_various(void)
+            }
+        }
+        return;
+    case VARIOUS_SET_OCTOLOCK:
+        if(gDisableStructs[gActiveBattler].octolock)
+        {
+            gBattlescriptCurrInstr = T1_READ_PTR(gBattlescriptCurrInstr + 3);
+        }
+        else
+        {
+            gDisableStructs[gActiveBattler].octolock = 1;
+            gBattleMons[gActiveBattler].status2 |= STATUS2_ESCAPE_PREVENTION;
+            gDisableStructs[gActiveBattler].battlerPreventingEscape = gBattlerAttacker;
+            gBattlescriptCurrInstr += 7;
+        }
+        return;
+    case VARIOUS_CHECK_POLTERGEIST:
+        if(gBattleMons[gActiveBattler].item == ITEM_NONE
+         || gFieldStatuses & STATUS_FIELD_MAGIC_ROOM
+         || GetBattlerAbility(gActiveBattler) == ABILITY_KLUTZ)
+        {
+            gBattlescriptCurrInstr = T1_READ_PTR(gBattlescriptCurrInstr + 3);
+        }
+        else
+        {
+            PREPARE_ITEM_BUFFER(gBattleTextBuff1, gBattleMons[gActiveBattler].item);
+            gBattlescriptCurrInstr += 7;
+        }
+        return;
+    case VARIOUS_TRY_NO_RETREAT:
+        if(gDisableStructs[gActiveBattler].noRetreat)
+            gBattlescriptCurrInstr = T1_READ_PTR(gBattlescriptCurrInstr + 3);
+        else
+        {
+            if(!(gBattleMons[gActiveBattler].status2 & STATUS2_ESCAPE_PREVENTION))
+                gDisableStructs[gActiveBattler].noRetreat = 1;
+            gBattlescriptCurrInstr += 7;
+        }
+        return;
+    case VARIOUS_TRY_TAR_SHOT:
+        if(gDisableStructs[gActiveBattler].tarShot)
+            gBattlescriptCurrInstr = T1_READ_PTR(gBattlescriptCurrInstr + 3);
+        else
+        {
+            gDisableStructs[gActiveBattler].tarShot = 1;
+            gBattlescriptCurrInstr += 7;
+        }
+        return;
+    case VARIOUS_TRY_NEXT_DART:
+        if (gBattleTypeFlags & BATTLE_TYPE_DOUBLE)
+        {
+            if ((gMoveResultFlags & MOVE_RESULT_MISSED || !IsBattlerAlive(BATTLE_PARTNER(gBattlerTarget)) || gMoveResultFlags & MOVE_RESULT_NO_EFFECT) && gMultiHitCounter == 1) //Dart hit a target, but missed the other OR no other target; attack 1st target again
+            {
+                gBattlerTarget = GetBattlerAtPosition(GetBattlerPosition(gBattlerTarget) ^ BIT_FLANK);
+                gBattlescriptCurrInstr = T1_READ_PTR(gBattlescriptCurrInstr + 3);
+            }
+            else if (gMultiHitCounter == 1) //Dart hit 1st target; move onto next
+            {
+                gBattlerTarget = GetBattlerAtPosition(GetBattlerPosition(gBattlerTarget) ^ BIT_FLANK);
+                gBattlescriptCurrInstr += 7;
+            }
+            else if ((gMoveResultFlags & MOVE_RESULT_MISSED || gMoveResultFlags & MOVE_RESULT_NO_EFFECT) && gMultiHitCounter == 2) //Dart missed 1st target
+            {
+                if(IsBattlerAlive(BATTLE_PARTNER(gBattlerTarget))) //If partner is alive, try to hit partner; otherwise, go to result
```

Comment:
```suggestion
            else if (gMultiHitCounter == 1) // Dart hit 1st target; move onto next
            {
                gBattlerTarget = GetBattlerAtPosition(GetBattlerPosition(gBattlerTarget) ^ BIT_FLANK);
                gBattlescriptCurrInstr += 7;
            }
            else if ((gMoveResultFlags & MOVE_RESULT_MISSED || gMoveResultFlags & MOVE_RESULT_NO_EFFECT) && gMultiHitCounter == 2) // Dart missed 1st target
            {
                if (IsBattlerAlive(BATTLE_PARTNER(gBattlerTarget))) // If partner is alive, try to hit partner; otherwise, go to result
```